src/coreclr/vm/virtualcallstub.cpp

```diff
@@ -1095,14 +1246,17 @@ BYTE *VirtualCallStubManager::GenerateStubIndirection(PCODE target, BOOL fUseRec
         POSTCONDITION(CheckPointer(RETVAL));
     } CONTRACT_END;
 
-    _ASSERTE(isStubStatic(target));
+    _ASSERTE(UseCachedInterfaceDispatch() || isStubStatic(target));
```

Comment:
this `UseCachedInterfaceDispatch()` doesn't look like it's ifdef guarded properly to ensure that the getter will be available, I'm guessing it doesn't cause a build failure because `_ASSERTE` evaluates to `(void)0` in release builds and cached interface dispatch is enabled in checked/debug?

Reply:
`UseCachedInterfaceDispatch()` isn't ifdef guarded.  it can be called in any build. Its simply hardcoded to false in those cases.

Reply:
I got confused here because there were two separate `UseCachedInterfaceDispatch`es and one was ifdef guarded